src/index.js

```diff
@@ -138,6 +158,20 @@ export default declare(({
           if (typeof filename === 'undefined' && typeof opts.filename !== 'string') {
             throw new TypeError('the "filename" option is required when transforming code');
           }
+
+          if (!path.scope.hasBinding('object.assign/implementation')) {
```

Comment:
but like, I don't think we want to polyfill people envs right?

Reply:
of course not :-) that's what "shim" does. "polyfill" is just a function that gets the best possible version.

Reply:
oh, ok, I'll change the code to use shim instead.. I went with the impl one cuz I can just call it, for shim I'll have to either somehow inject this expr:

```js
var objectAssignShim = require('object.assign/shim');
 // it probably needs to be separated for cases where we generate `import` syntax... 🤔 but if you think it is really safe to call require(...)(); I'm up for it
var objectAssign = objectAssignShim();

var Svg = function(overrides) {
   var props = objectAssign({}, DEFAULTS, overrides);
   // ....
}
```

or make it like this, which is the easy way to do it...

```js
var objectAssignShim = require('object.assign/shim');

var Svg = function(overrides) {
   var props = objectAssignShim()({}, DEFAULTS, overrides);
   // ....
}
```

I'll make it like this ^, I don't really know the implications of that `objectAssignShim()` call on every render.. but I don't think it will be simple to inject that `var objectAssign = objectAssignShim()` at the top levels

Reply:
Made the change

Reply:
sorry if i was unclear; shim is the one we DONT want to use. polyfill is the one we DO want to use, because it will be the native one when possible.

Reply:
ohhh, sorry for misunderstanding, I thought polyfill would polyfill the env 😆 (now what I read bout shim makes more sense) Ill make the changes tomorrow

Reply:
made the change to use polyfill